### PR TITLE
freeze_backbone and freeze_decoder docstrings update

### PR DESCRIPTION
**Pull request by isaaccorley**
This PR updates missing docstrings for the freeze parameters.

### CODE REVIEW

File: torchgeo/trainers/classification.py [hunk 1]
```diff
@@ -97,13 +97,14 @@ def __init__(self, **kwargs: Any) -> None:
...
             learning_rate: Learning rate for optimizer
             learning_rate_schedule_patience: Patience for learning rate scheduler
-            freeze_backbone: Fine-tune the cls head by freezing the backbone
+            freeze_backbone: Freeze the backbone network to linear probe
+                the classifier head
 
         .. versionchanged:: 0.4
            The *classification_model* parameter was renamed to *model*.
 
         .. versionchanged:: 0.5
```
Comment (adamjstewart) [inline torchgeo/trainers/classification.py:106]:
```suggestion
        .. versionadded:: 0.5
```

File: torchgeo/trainers/classification.py [hunk 2]
```diff
@@ -302,9 +303,14 @@ def __init__(self, **kwargs: Any) -> None:
             in_channels: Number of input channels to model
             learning_rate: Learning rate for optimizer
             learning_rate_schedule_patience: Patience for learning rate scheduler
+            freeze_backbone: Freeze the backbone network to linear probe
+                the classifier head
 
         .. versionchanged:: 0.4
            The *classification_model* parameter was renamed to *model*.
+
+        .. versionchanged:: 0.5
```
Comment (adamjstewart) [inline torchgeo/trainers/classification.py:312]:
```suggestion
        .. versionadded:: 0.5
```

File: torchgeo/trainers/detection.py [hunk 3]
```diff
@@ -178,7 +178,7 @@ def __init__(self, **kwargs: Any) -> None:
            The *detection_model* parameter was renamed to *model*.
 
         .. versionchanged:: 0.5
```
Comment (adamjstewart) [inline torchgeo/trainers/detection.py:180]:
```suggestion
        .. versionadded:: 0.5
```

File: torchgeo/trainers/regression.py [hunk 4]
```diff
@@ -93,13 +93,16 @@ def __init__(self, **kwargs: Any) -> None:
...
-            freeze_backbone: Fine-tune the cls head by freezing the backbone
+            freeze_backbone: Freeze the backbone network to linear probe
+                the regression head
+            freeze_decoder: Freeze the decoder network to linear probe
+                the regression head
 
         .. versionchanged:: 0.4
             Change regression model support from torchvision.models to timm
 
         .. versionchanged:: 0.5
```
Comment (adamjstewart) [inline torchgeo/trainers/regression.py:104]:
```suggestion
        .. versionadded:: 0.5
```

File: torchgeo/trainers/segmentation.py [hunk 5]
```diff
@@ -111,6 +111,10 @@ class and used with 'ce' loss
             ignore_index: Optional integer class index to ignore in the loss and metrics
             learning_rate: Learning rate for optimizer
             learning_rate_schedule_patience: Patience for learning rate scheduler
+            freeze_backbone: Freeze the backbone network to fine-tune the
+                decoder and segmentation head
+            freeze_decoder: Freeze the decoder network to linear probe
+                the segmentation head
```
Comment (adamjstewart) [inline torchgeo/trainers/segmentation.py:117]: Needs versionadded

Comment (calebrob6) [reply torchgeo/trainers/segmentation.py:117]: I think this is done right?

File: torchgeo/trainers/regression.py [hunk 6]
```diff
@@ -93,13 +93,16 @@ def __init__(self, **kwargs: Any) -> None:
             in_channels: Number of input channels to model
             learning_rate: Learning rate for optimizer
             learning_rate_schedule_patience: Patience for learning rate scheduler
-            freeze_backbone: Fine-tune the cls head by freezing the backbone
+            freeze_backbone: Freeze the backbone network to linear probe
+                the regression head
+            freeze_decoder: Freeze the decoder network to linear probe
+                the regression head
```
Comment (adamjstewart) [inline torchgeo/trainers/regression.py:99]: Should we clarify that this only supports unet and deeplabv3+?

Comment (adamjstewart) [reply torchgeo/trainers/regression.py:99]: And that this is only used by PixelwiseRegression?

Comment (adamjstewart) [reply torchgeo/trainers/regression.py:99]: Just waiting on this update

File: torchgeo/trainers/regression.py [hunk 7]
```diff
@@ -93,13 +93,17 @@ def __init__(self, **kwargs: Any) -> None:
             in_channels: Number of input channels to model
             learning_rate: Learning rate for optimizer
             learning_rate_schedule_patience: Patience for learning rate scheduler
-            freeze_backbone: Fine-tune the cls head by freezing the backbone
+            freeze_backbone: Freeze the backbone network to linear probe
+                the regression head
```
Comment (adamjstewart) [inline torchgeo/trainers/regression.py:97]: Same comment for backbone

Comment (isaaccorley) [reply torchgeo/trainers/regression.py:97]: If you can finish this that would be great.

Comment (isaaccorley) [reply torchgeo/trainers/regression.py]: Nvm it's updated now

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**